zenith_utils/src/control_file.rs

```diff
@@ -0,0 +1,167 @@
+use crate::bin_ser::BeSer;
+use anyhow::{bail, ensure, Result};
+use serde::{de::DeserializeOwned, Serialize};
+use std::{
+    convert::TryInto,
+    fs::{File, OpenOptions},
+    io::Write,
+    path::Path,
+};
+
+// Taken from PG_CONTROL_MAX_SAFE_SIZE
+pub const MAX_CONTROL_SAFE_SIZE: usize = 512;
+const CHECKSUM_SIZE: usize = std::mem::size_of::<u32>();
+pub const MAX_CONTROL_DATA_SIZE: usize = MAX_CONTROL_SAFE_SIZE - CHECKSUM_SIZE;
+
+// this trait serves as a proof that size of the implementing struct is less than MAX_CONTROL_DATA_SIZE
+// see IsSafeControlFile macro at the end of the file
```

Comment:
Could you make this a doc comment? Then you'd be able to reference `[IsSafeControlFile]` etc.